scripts/AutoRebalance.py

```diff
@@ -0,0 +1,499 @@
+        "BNB": Decimal('10.00'),
+        "AVAX": Decimal('5.00'),
+        "MATIC": Decimal('5.00')
+    }
+    coin_config = ut_coin_config  # Initialize coin_config
+
+    # Set rebalance threshold
+    ut_threshold = Decimal('2.00')
+    dt_threshold = Decimal('1.00')
+    threshold = dt_threshold  # Initialize threshold
+
+    # Abstract coin name and Make coin list
+    hold_asset = abstract_keys(hold_asset_config)[0]
+    coins = abstract_keys(coin_config)
+    pairs = make_pairs(coins, hold_asset)
+
+    # Put connector and pairs into markets
+    markets = {connector_name: pairs}
+
+    # Set status
+    status = "rebalancing"
+
+    # Set klines configuration
+    last_data_ts = 0.
+    data_interval = 900  # Should be equal to interval
+    interval = "15m"
+    limit = 60  # Should be more than l_atr_period
+
+    # Set long and short atr configuration
+    s_atr_period = 13
+    l_atr_period = 34
+    s_atr_list = []  # Placeholder
+    l_atr_list = []  # Placeholder
+    s_atr_mean = 0.  # Placeholder
+    l_atr_mean = 0.  # Placeholder
+    is_volatile = True  # Initialize is_volatile
+
+    # Set SuperTrend configuration
+    trend_atr_period = 13
+    atr_mult = 3.
+    upperband = 0.  # Placeholder
+    lowerband = 0.  # Placeholder
+    preclose = 0.  # Placeholder
+    supertrend_atr = 0.  # Placeholder
+    is_uptrend = True  # Initialize is_uptrend
+    _supertrend = pd.DataFrame()  # initialize _supertrend df
+
+    # Set active order or not
+    mm_mode = True
+
+    # Set user define market making or not
+    udefine_mm = True
+
+    # Set market making threshold
+    mm_threshold = Decimal('0.50')
+
+    @property
+    def connector(self):
+        """
+        The only connector in this strategy, define it here for easy access
+        """
+        return self.connectors[self.connector_name]
+
+    def on_tick(self):
+        exchange = self.connector
+
+        if self.last_data_ts < (self.current_timestamp - self.data_interval):
+            self.logger().info("Time to get new data !!!")
+            df_all = pd.DataFrame()
+
+            for coin in self.coins:
+                pair = coin + "-" + self.hold_asset
+                all_df = get_klines(pair, self.interval, self.limit)
+
+                # ATR indicator
+                # Calculate long and short ATR
+                all_df['s_atr'] = atr(all_df, self.s_atr_period)
+                s_atr = all_df.iloc[-1]['s_atr']
+                self.s_atr_list.append(s_atr)
+                all_df['l_atr'] = atr(all_df, self.l_atr_period)
+                l_atr = all_df.iloc[-1]['l_atr']
+                self.l_atr_list.append(l_atr)
+
+                # Prepare df for supertrend
+                s_df = all_df.select_dtypes(exclude=["datetime"])
+                df_all = df_all.add(s_df, fill_value=0)
+                all_df[s_df.columns] = s_df.add(df_all)
+                self._supertrend = supertrend(all_df, self.trend_atr_period, self.atr_mult)
```

Comment:
you are overriding this variable for each trading pair, so the last pair is only to be considered

Reply:
fixed